CSharp/Library/Microsoft.Bot.Connector.Shared/MicrosoftAppCredentials.cs

```diff
@@ -244,33 +263,32 @@ private OAuthException(SerializationInfo info, StreamingContext context)
 #endif
         }
 
+        private static HttpClient g_httpClient = new HttpClient();
+
         private async Task<OAuthResponse> RefreshTokenAsync()
         {
-            using (HttpClient httpClient = new HttpClient())
-            {
-                var content = new FormUrlEncodedContent(new Dictionary<string, string>()
+            var content = new FormUrlEncodedContent(new Dictionary<string, string>()
                 {
                     { "grant_type", "client_credentials" },
                     { "client_id", MicrosoftAppId },
                     { "client_secret", MicrosoftAppPassword },
                     { "scope", OAuthScope }
                 });
 
-                using (var response = await httpClient.PostAsync(OAuthEndpoint, content).ConfigureAwait(false))
+            using (var response = await g_httpClient.PostAsync(OAuthEndpoint, content).ConfigureAwait(false))
+            {
+                string body = null;
+                try
                 {
-                    string body = null;
-                    try
-                    {
-                        response.EnsureSuccessStatusCode();
-                        body = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-                        var oauthResponse = JsonConvert.DeserializeObject<OAuthResponse>(body);
-                        oauthResponse.expiration_time = DateTime.UtcNow.AddSeconds(oauthResponse.expires_in).Subtract(TimeSpan.FromSeconds(60));
-                        return oauthResponse;
-                    }
-                    catch (Exception error)
-                    {
-                        throw new OAuthException(body ?? response.ReasonPhrase, error);
-                    }
+                    response.EnsureSuccessStatusCode();
+                    body = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                    var oauthResponse = JsonConvert.DeserializeObject<OAuthResponse>(body);
+                    oauthResponse.expiration_time = DateTime.UtcNow.AddSeconds(oauthResponse.expires_in).Subtract(TimeSpan.FromSeconds(60));

```

Comment:
This pattern can really confuse things. Instead of storing the actual expiration, this code recalculates what the refresh time should be, and then stores that as if it were the expiration.

I think it should just store the raw expiration and if we're going to do any calculation, we can do when we read the value later.